spec/test_spec.lua

```diff
@@ -70,6 +70,30 @@ describe("luarocks test #integration", function()
          -- Assert that busted ran, whether successfully or not
          assert.match("%d+ success.* / %d+ failure.* / %d+ error.* / %d+ pending", output)
       end)
+
+      it("prepare", function()
+         finally(function()
+            -- delete downloaded and unpacked files
+            lfs.chdir(testing_paths.testrun_dir)
+            test_env.remove_dir("busted_project-0.1-1")
+            os.remove("busted_project-0.1-1.src.rock")
+         end)
+
+         -- make luassert
+         assert.is_true(run.luarocks_bool("download --server="..testing_paths.fixtures_repo_dir.." busted_project 0.1-1"))
+         assert.is_true(run.luarocks_bool("unpack busted_project-0.1-1.src.rock"))
+         lfs.chdir("busted_project-0.1-1/busted_project")
+         assert.is_true(run.luarocks_bool("make"))
+         assert.is_true(run.luarocks_bool("show busted"))
+         local prepareOutput = run.luarocks_bool("test --prepare")
```

Comment:
```lua
assert.is_true(run.luarocks_bool("remove busted"))
local prepareOutput = run.luarocks_bool("test --prepare")
assert.is_true(run.luarocks_bool("show busted"))
``` 
The assertion will fail because before the ```test --prepare``` busted is not installed. So it will throw an error could not find rock 'busted'. After ```test --prepare``` we can remove busted because busted is install by ```test --prepare```.
```lua
local prepareOutput = run.luarocks_bool("test --prepare")
assert.is_true(run.luarocks_bool("show busted"))
assert.is_true(run.luarocks_bool("remove busted"))

Reply:
That is strange! How can `show busted` before `test --prepare` pass (meaning that `busted` was installed) and `remove busted` before `test --prepare` fail (maining that `busted` was not installed)?

If we can't figure this out, I suggest not spending too much time on it since that is not the main goal of the test. So, this should be sufficient:

```lua
-- ensure busted is not there before running `test --prepare`
-- we don't care if it was installed or not
run.luarocks_bool("remove busted")
local prepareOutput = run.luarocks_bool("test --prepare")
-- ensure busted *is* there after running `test --prepare`
-- meaning that it did install it
assert.is_true(run.luarocks_bool("show busted"))
```

Reply:
(Ah, I think I have a possible cause to what's going on — the test suite runs twice, once in a "minimal" environment with nothing pre-installed and once in a "full" environment with a bunch of rocks pre-installed... this is what may be causing different results. Anyway, this is just a side comment and doesn't change my suggestion above!)